src/Mod/Draft/Draft.py

```diff
@@ -791,7 +792,7 @@ def makeRectangle(length, height, placement=None, face=None, support=None):
         FreeCAD.Console.PrintError("No active document. Aborting\n")
         return
     if placement: typecheck([(placement,FreeCAD.Placement)], "makeRectangle")
-    obj = FreeCAD.ActiveDocument.addObject("Part::Part2DObjectPython","Rectangle")
+    obj = FreeCAD.ActiveDocument.addObject("Part::Part2DObjectPython",name)
```

Comment:
I don't think you should change the objects Name. You should just alter the object's Label. This is mentioned in the forum thread: [Extend Draft.make### functions by a name argument?](https://forum.freecadweb.org/viewtopic.php?p=325439#p325439)

```
    obj = FreeCAD.ActiveDocument.addObject("Part::Part2DObjectPython", "Rectangle")  # Fixed name
    obj.Label = name  # variable label
```

The Label is the user editable field, while the Name stays fixed through the duration of the object's life. A user can do everything with the Label, while the Name is more of an internal string.

Reply:
Well, that would be a bit inconsistent, in particular since
```
makeArray
makeSketch
makePoint
makeFacebinder
```
already allow to set a name.

But I don't mind.
It's fine with me if the pull request is not merged.
It was just a suggestion for improvement.

I can easily write my own wrapper functions.
I need to be able to change the name and not just the label.

Reply:
Yeah, but maybe those functions should be changed, and not all others.

Yorik mentions that the Name has problems with non-ASCII characters, so keeping the Name fixed seems to be better. Maybe you should answer Yorik's question: [In what kind of scenario would you like to have the object named as something else?](https://forum.freecadweb.org/viewtopic.php?p=326009#p326009)

Reply:
Indeed this lead to a lot of problems in the past, specially when "name" is made of no ASCII characters (ex, chinese or arabic characters)... In that case the name ends up being something like ______001. Also, the name of an object is not really something to be used directly by users, it's more like an ID string. So it makes sense to keep it to something that can be more or less universally understood. The makeArray, makeSketch, etc... are actually leftovers from the past and should be fixed too.

I'm going to close this PR, okay? If you feel that's not right reopen and we discuss. Maybe the proper way to do this would be to add another argument to the make### function, like makeWall([other args],name="Wall",label="Wall")

Reply:
No worries!